security/fma-holocene.md

```diff
@@ -0,0 +1,117 @@
+- [Update to the MIPS contract](#update-to-the-mips-contract)
+- [Generic Items](#generic-items)
+- [Audit Requirements](#audit-requirements)
+- [Action Items](#action-items)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+| | |
+|--------|--------------|
+| Author | George Knee |
+| Created at | 2024-09-24 |
+| Needs Approval From | maurelian |
+| Other Reviewers |   |
+| Status | Draft |
+
+
+## Introduction
+
+This document is intended to be shared in a public space for reviews and visibility purposes. It covers the Holocene hardfork, which involves the following changes: 
+- (Consensus Layer) Holocene Derivation
+- (Smart Contracts) Deterministic Standard L2 Genesis
+- (Execution Layer & Smart Contracts) Configurable EIP-1559 Parameters via SystemConfig
+- (Execution Layer) L2ToL1MessagePasser Storage Root in Header
+- (Smart Contracts) Update to the MIPS contract
+
+Each change has it's own section below with a list of Failure Modes.
+
+Below are references for this project:
+
+- [PID: Holocene hardfork upgrade](https://www.notion.so/PID-Holocene-hardfork-upgrade-00ee1ffc414a407088fdb49841771527?pvs=21)
+- [Github tracker](https://github.com/orgs/ethereum-optimism/projects/84/views/6)
+- [Specs](https://specs.optimism.io/protocol/holocene/derivation.html?highlight=holocene#holocene-derivation)
+
+
+##  Holocene Derivation
+
+### The batcher violates the new (stricter) ordering rules
+
+#### Description
+
+If the batcher:
+  - sends transactions out of order, or
+   - orders frames within a transaction out of order
+
+This will lead to the consensus client hitting error conditions when it loads the frames from L1. According to the spec, frames will be dropped in order to maintain a contiguous frame queue. If those frames are not re-sent in a timely manner on L1, the safe chain would halt.
+    
+#### Risk Assessment
+ medium severity / medium likelihood
+#### Mitigations
+The batcher implementation could:
+  - leverage nonce management to avoid sending transactions out of order in the first place (see [spec](https://specs.optimism.io/protocol/holocene/derivation.html?highlight=holocene#batcher-hardening))
+  - be hardened so detect the chain halt and resubmit the frames which were dropped (see [spec](https://specs.optimism.io/protocol/holocene/derivation.html?highlight=holocene#batcher-hardening)) either:
+    - as a part of normal operation
+    - as a part of its startup behavior (i.e. to be triggered by a restart)
+    - as a part of an emergency mode triggered by an admin API, this could allow for manual intervention
+- continually check for contiguity in its internal state, and panic if this is violated (possibly then triggering some recovery behavior)
+#### Detection
+L2 safe chain halt
+#### Recovery Path(s)
+We would need to get the batcher to resubmit the transaction sent out of order. No governance or hardfork needed to recover.
```

Comment:
A simple batcher restart would cause the batcher to re-batch the unsafe chain, which should lead to recovery. Moreover, we could also temporarily operate the batcher with reduced tx sending concurrency, which should avoid out of order txs.